pkg/resourcecreator/pod/pod.go

```diff
@@ -105,33 +105,35 @@ func CreateSpec(ast *resource.Ast, cfg Config, appName string, annotations map[s
 		TerminationGracePeriodSeconds: terminationGracePeriodSeconds,
 	}
 
-	if cfg.IsSecurePodSecurityContextEnabled() && !exploitable(annotations) { // TODO(jhrv): remove SecurePodSecurityContext option all together when this is rolled out in all clusters
-		podSpec.Containers[0].SecurityContext = &corev1.SecurityContext{
-			RunAsUser:                pointer.Int64Ptr(runAsUser(annotations)),
-			RunAsGroup:               pointer.Int64Ptr(runAsGroup(annotations)),
-			RunAsNonRoot:             pointer.BoolPtr(true),
-			Privileged:               pointer.BoolPtr(false),
-			AllowPrivilegeEscalation: pointer.BoolPtr(false),
-			ReadOnlyRootFilesystem:   pointer.BoolPtr(readOnlyFileSystem(annotations)),
-		}
+	podSpec.Containers[0].SecurityContext = &corev1.SecurityContext{
+		RunAsUser:                pointer.Int64Ptr(runAsUser(annotations)),
+		RunAsGroup:               pointer.Int64Ptr(runAsGroup(annotations)),
+		RunAsNonRoot:             pointer.BoolPtr(true),
+		Privileged:               pointer.BoolPtr(false),
+		AllowPrivilegeEscalation: pointer.BoolPtr(false),
+		ReadOnlyRootFilesystem:   pointer.BoolPtr(readOnlyFileSystem(annotations)),
+	}
 
-		capabilities := &corev1.Capabilities{
-			Drop: []corev1.Capability{"all"},
-		}
+	capabilities := &corev1.Capabilities{
+		Drop: []corev1.Capability{"all"},
+	}
 
-		additionalCapabilities := sanitizeCapabilities(annotations, cfg.GetAllowedKernelCapabilities())
-		if len(additionalCapabilities) > 0 {
-			capabilities.Add = additionalCapabilities
-		}
+	additionalCapabilities := sanitizeCapabilities(annotations, cfg.GetAllowedKernelCapabilities())
+	if len(additionalCapabilities) > 0 {
+		capabilities.Add = additionalCapabilities
+	}
+
+	podSpec.Containers[0].SecurityContext.Capabilities = capabilities
 
-		podSpec.Containers[0].SecurityContext.Capabilities = capabilities
+	if exploitable(annotations) && !cfg.IsSecurePodSecurityContextEnforced() {
```

Comment:
Det at denne kommer her nede, etter alt arbeidet for å sette opp security context tidligere i funksjonen er litt rart.
Kunne man heller trukket security context koden ut i en egen funksjon, og gjort early return i dette tilfellet?

Reply:
enig, fixed